src/Build/Globbing/MSBuildGlob.cs

```diff
@@ -202,8 +206,20 @@ public static MSBuildGlob Parse(string globRoot, string fileSpec)
 
                     if (regex == null)
                     {
+                        RegexOptions regexOptions = FileMatcher.DefaultRegexOptions;
                         // compile the regex since it's expected to be used multiple times
-                        Regex newRegex = new Regex(matchFileExpression, FileMatcher.DefaultRegexOptions | RegexOptions.Compiled);
+                        // For the kind of regexes used here, compilation on .NET Framework tends to be expensive and not worth the small
+                        // run-time boost so it's enabled only on .NET Core by default.
+#if RUNTIME_TYPE_NETCORE
+                        bool compileRegex = true;
+#else
```

Comment:
These aren't things I would expect to change in the middle of a build. Should we save off the RegexOptions earlier and use it multiple times?

Reply:
Checking the changewave and OR'ing two integers is very cheap. Unless I am misunderstanding I would argue that it's not worth adding a cache for this.